15.js

```diff
@@ -0,0 +1,39 @@
+const readline = require('readline')
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+})
+
+let state = 'CLOSED'
+
+console.log('\nFinite State Gate')
+console.log("Enter 'remoteClicked' or 'cycleComplete'\n")
+
+console.log('Gate:', state)
+
+rl.on('line', input => {
```

Comment:
Even though this is a fairly straightforward implementation, there's a lot going on in this block.  Consider refactoring to isolate some of the logically separable behavior.